src/app/ui/dialog/dialog-container/dialog-container.component.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  Component,
+  OnInit,
+  ChangeDetectionStrategy,
+  HostListener,
+  ElementRef,
+  Output,
+  EventEmitter,
+  Renderer,
+  AfterViewInit,
+  Input
+} from '@angular/core';
+
+export interface DialogOptions {
+  closeOnEsc?: boolean;
+  closeOnClickOutside?: boolean;
+}
+
+@Component({
+  selector: 'iw-dialog-container',
+  templateUrl: './dialog-container.component.html',
+  styleUrls: ['./dialog-container.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DialogContainerComponent implements OnInit, AfterViewInit {
+  @Input() dialogOptions: DialogOptions;
```

Comment:
You should provide an object with default values since you're accessing `this.dialogOptions.closeOnEsc` . 
Typescript compiler option `strictNullChecks` helps with this kind of errors.